### PR TITLE
Typescript: Preserve decorated definite class properties

### PR DESCRIPTION
**Pull request by jamescdavis**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            |
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

This is a follow-up to #8238, but for class properties marked as definite (using the non-null assertion `!`)

### Current behavior:
The `babel-plugin-transform-typescript` removes class properties marked as definite regardless if decorators are assigned to it or not.

```ts
// Original
class C {
  @foo
  d;
  @foo
  e = 3;
  @foo
  f!;
}

// Transformed
class C {
  @foo
  d;
  @foo
  e = 3;
  // <-- Missing `f` field
}
```

### Expected behavior:
The babel-plugin-transform-typescript should only remove a definite class property if it has no decorators.

```ts
// Original
class C {
  @foo
  d;
  @foo
  e = 3;
  @foo
  f!;
}

// Transformed
class C {
  @foo
  d;
  @foo
  e = 3;
  @foo
  f;
}
```

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
Thanks!

File: packages/babel-plugin-transform-typescript/src/index.js [hunk 1]
```diff
@@ -74,7 +74,11 @@ export default declare(
             );
           }
 
-          path.remove();
+          if (node.definite && node.decorators) {
+            node.definite = null;
+          } else {
+            path.remove();
+          }
```
Comment (JLHwung) [inline packages/babel-plugin-transform-typescript/src/index.js]: nit: I feel like it's clearer to only check against `!node.decorators` here (like we did in the `else if` branch).
```js
if (!node.decorators) {
  path.remove();
}
```
and then reset `definite` later (like we did for `abstract`, `accessibility`, etc.)
```js
if (node.definite) node.definite = null;
```

Comment (jamescdavis) [reply packages/babel-plugin-transform-typescript/src/index.js]: that's a *definite* improvement in clarity 😉

**Comment by JLHwung** [comment]
@jamescdavis Can you rebase on upstream master? The e2e error will be fixed after rebasing.

**Pull request merged**